src/cpp/server/server_cc.cc

```diff
@@ -934,6 +1006,13 @@ void Server::ShutdownInternal(gpr_timespec deadline) {
       (*it)->Wait();
     }
 
+    // Wait for all outstanding callback requests to complete
```

Comment:
Apparently I can only comment on changed lines, but I wanted to add my nit:
At the beginning of this block, can we do a if (!shutdown_) return; to avoid one level of nesting?